presto-hive/src/main/java/io/prestosql/plugin/hive/orc/OrcPageSourceFactory.java

```diff
@@ -191,7 +191,9 @@ private static OrcPageSource createOrcPageSource(
         AggregatedMemoryContext systemMemoryUsage = newSimpleAggregatedMemoryContext();
         try {
             OrcReader reader = new OrcReader(orcDataSource, options);
-
+            if (reader.isEmpty()) {
```

Comment:
Looks like this would also have to handle handle the case where split start + length exceeds the discovered file size as well as the rarer case where start >= discovered file size.

Reply:
ORC uses a similar system to parquet.  The split region is basically a selection for ORC stripes.  Only stripes that start within the split region are considered part of the split, so if the split region is off the end of the file, there will be no blocks in that region.

Reply:
Got it, makes sense that the check isn't necessary for ORC and Parquet then.